WordPress/Classes/Utility/Ratings/AppRatingUtility.m

```diff
@@ -1,38 +1,97 @@
 NSString *const AppRatingSignificantEventCount = @"AppRatingSignificantEventCount";
 NSString *const AppRatingUseCount = @"AppRatingUseCount";
+NSString *const AppRatingNumberOfVersionsSkippedPrompting = @"AppRatingsNumberOfVersionsSkippedPrompt";
+NSString *const AppRatingNumberOfVersionsToSkipPrompting = @"AppRatingsNumberOfVersionsToSkipPrompting";
 NSString *const AppRatingRatedCurrentVersion = @"AppRatingRatedCurrentVersion";
 NSString *const AppRatingDeclinedToRateCurrentVersion = @"AppRatingDeclinedToRateCurrentVersion";
 NSString *const AppRatingGaveFeedbackForCurrentVersion = @"AppRatingGaveFeedbackForCurrentVersion";
+NSString *const AppRatingDislikedCurrentVersion = @"AppRatingDislikedCurrentVersion";
+NSString *const AppRatingLikedCurrentVersion = @"AppRatingLikedCurrentVersion";
+NSString *const AppRatingUserLikeCount = @"AppRatingUserLikeCount";
+NSString *const AppRatingUserDislikeCount = @"AppRatingUserDislikeCount";
 
-+ (BOOL)shouldPromptForAppReview
+NSString *const AppReviewPromptDisabledUrl = @"http://api.wordpress.org/iphoneapp/app-review-prompt-check/1.0/";
+
+- (instancetype)init
 {
-    if ([self interactedWithAppReviewPrompt]) {
-        return NO;
-    }
-    
-    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-    NSUInteger significantEventCount = [userDefaults integerForKey:AppRatingSignificantEventCount];
-    NSUInteger numberOfSignificantEventsRequiredForPrompt = [userDefaults integerForKey:AppRatingNumberOfSignificantEventsRequiredForPrompt];
-    
-    if (significantEventCount >= numberOfSignificantEventsRequiredForPrompt) {
-        return YES;
+    self = [super init];
+    if (self) {
+        _sections = [NSMutableDictionary dictionary];
+        _disabledSections = [NSMutableDictionary dictionary];
     }
+    return self;
+}
+
++ (instancetype)sharedInstance {
+    static AppRatingUtility *_sharedAppRatingUtility = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sharedAppRatingUtility = [self new];
+    });
     
-    return NO;
+    return _sharedAppRatingUtility;
 }
 
-+ (BOOL)interactedWithAppReviewPrompt
++ (void)resetReviewPromptDisabledStatus
 {
-    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-    return [userDefaults boolForKey:AppRatingRatedCurrentVersion] || [userDefaults boolForKey:AppRatingDeclinedToRateCurrentVersion] || [userDefaults boolForKey:AppRatingGaveFeedbackForCurrentVersion];
+    AppRatingUtility *appRatingUtility = [AppRatingUtility sharedInstance];
+    appRatingUtility.allPromptingDisabled = NO;
+    [appRatingUtility.disabledSections enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+        appRatingUtility.disabledSections[key] = @(NO);
+    }];
+}
+
++ (void)checkIfAppReviewPromptsHaveBeenDisabled:(void (^)(void))success failure:(void (^)(void))failure;
```

Comment:
Again, minor... but if you:

`typedef void (^AppRatingCompletionBlock)();`

You can then:

`+ (void)checkIfAppReviewPromptsHaveBeenDisabled:(AppRatingCompletionBlock)success failure:(AppRatingCompletionBlock)failure;

Which also lets callers use:

`AppRatingCompletionBlock success = ^void() {...}`

Feel free to opt out of this if you don't like it though!